src/osvr/Client/ImagingRemoteFactory.cpp

```diff
@@ -88,8 +88,8 @@ namespace client {
             report.state.metadata = data.metadata;
             report.state.data = data.buffer.get();
             for (auto &iface : m_interfaces) {
-                iface->triggerCallbacks(timestamp, report);
                 iface->getContext().acquireObject(data.buffer);
+                iface->triggerCallbacks(timestamp, report);
```

Comment:
Just occurred to me - the only case in which this would not be good is if somehow the list of interfaces was populated but there were no callbacks registered. We probably need to handle that case.